app/sprinkles/account/assets/userfrosting/js/uf-password-security.js

```diff
@@ -0,0 +1,45 @@
+$(document).ready(function() {
```

Comment:
There is a potential error in here (a function will be called despite not being defined if no compromised passwords is disabled). Also, try to keep string quotes consistent. There isn't any linting for this currently, but we should strive for consistency.

Logic here should be something like this.

```js
$(document).ready(function() {

    var breachThreshold = site.password.security.enforce_no_compromised.breaches;

    // Only perform security checks if they are enabled
    if (breachThreshold != -1) {
        // Setup before functions
        var passwordInput = $('form"'.find('input[name=password]');
        var typingTimer; // Timer identifier
        var doneTypingInterval = 200; // Time in ms

        /**
         * Checks if password is valid according to site policies.
         */
        function triggerPasswordValidation() {
            var ajaxData = {};
            ajaxData['password'] = $(passwordInput).val();
            ajaxData[site.csrf.keys.name] = site.csrf.name;
            ajaxData[site.csrf.keys.value] = site.csrf.value;
            $.ajax({
                type: 'POST',
                url: site.uri.public + '/account/check-password',
                data: ajaxData,
                success: function(data) {

                    $(passwordInput).rules('add', {
                        'check-password': data
                    });

                    $(passwordInput).valid();
                }
            });
        }

        // On keyup, start the countdown
        passwordInput.keyup(function() {
            clearTimeout(typingTimer);
            if (passwordInput.val()) {
                typingTimer = setTimeout(triggerPasswordValidation, doneTypingInterval);
            }
        });


        jQuery.validator.addMethod(
            'check-password',
            function(value, element, breaches) {
                return breaches <= breachThreshold;
            },
            jQuery.validator.format('The password security policy does not allow use of passwords that have been exposed by a prior data breach. Please enter a different password.')
        );

    }
});
```

Reply:
lol... I'm going to quote you when this was inside a conditional, as you're suggesting it be again: 

> I can't see any reason for this to be wrapped in a conditional. Currently I can see this cause JS errors under certain scenarios as well.

Reply:
Fixed in e985aee4d80605c1684bb2708385dfea15763378